aten/src/ATen/native/cuda/jit_utils.cpp

```diff
@@ -651,8 +690,9 @@ std::string generate_code(
   } else {
     env.s("bfloat16_string", "");
   }
-  if (dynamic_casting) {
-    env.s("complex_string", jiterator_complex_support_literal);
+  if (f_inputs_type == "std::complex<float>" || result_type == "std::complex<float>" ||
+      f_inputs_type == "std::complex<double>" || result_type == "std::complex<double>"|| dynamic_casting) {
+    env.s("complex_string", get_complex_string());
```

Comment:
Should we add a check that there are no complex half inputs, or that the compute type is not complex half, in Loops.cuh?